src/Microsoft.DotNet.Cli.Utils/CommandResolver.cs

```diff
@@ -0,0 +1,191 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using NuGet.Frameworks;
+using System.Linq;
+using System.Runtime.InteropServices;
+using Microsoft.DotNet.ProjectModel;
+using Microsoft.DotNet.ProjectModel.Graph;
+using NuGet.Packaging;
+
+namespace Microsoft.DotNet.Cli.Utils
+{
+    internal static class CommandResolver
+    {
+        public static ProcessStartInfo TryResolvePathAndArgs(string commandName, string args, NuGetFramework framework = null)
+        {
+            return ResolveFromRootedCommand(commandName, args) ??
+                   ResolveFromProjectDependencies(commandName, args, framework) ??
+                   ResolveFromProjectTools(commandName, args) ??
+                   ResolveFromPath(commandName, args);
```

Comment:
Is it the case that all the clients of Command.Create need all the resolvers?
If not, I'd suggest converting the different ResolveFrom* into multiple strategies / chains of command and have the client select which resolver chain configuration to use. (either clients assemble the resolvers like Lego pieces, or Command has some predefined resolver chains that clients can use). This would save some computation and document in code which clients use which resolver sets.